### PR TITLE
PIL-1023: vNext develop merge

### PR DESCRIPTION
**Pull request by poocart**
https://linear.app/pillarproject/issue/PIL-1023/merge-develop-etherspot-with-vnext-v2

This PR includes changes needed for `vNext` to update with latest develop that contains `Etherspot`.

Note: @mdjastrzebski I wonder if `useSmartWalletType` should be renamed to useAccountType ant combined with account types, wdyt? Also would be great if you could build from this branch to check visuals if nothing is off 👀

### CODE REVIEW

**Review by mdjastrzebski — APPROVED** [review]
Overall looks great 🚀 🌔 

We should fill the body of useSmartWalletType (-> useActiveAccountType) and move it to `selectors.js`

Otherwise, there are some potential edge-cases for viewTransactionOnBlockchain that might require some extra attention.

File: src/utils/blockchainExplorer.js [hunk 1]
```diff
@@ -24,11 +24,25 @@ import t from 'translations/translate';
...
+import archanovaService from 'services/archanova';
+import etherspotService from 'services/etherspot';
+
+// Types
+import type { Accounts } from 'models/Account';
+import type { Event } from 'models/History';
 
-export function viewOnBlockchain(hash: ?string) {
-  if (!hash) {
+export async function viewTransactionOnBlockchain(event: Event, accounts: Accounts) {
```
[the remark this answers is not in the document]
Comment (poocart) [reply src/utils/blockchainExplorer.js:42]: Makes sense 👍

File: src/utils/blockchainExplorer.js [hunk 2]
```diff
@@ -24,11 +24,25 @@ import t from 'translations/translate';
...
+  const { hash = null, batchHash = null, fromAddress = null } = event;
```
[the remark this answers is not in the document]
Comment (poocart) [reply src/utils/blockchainExplorer.js:43]: `batchHash` is temporary thing which can happen ONLY to outgoing transactions while those are in a transfer state from batch to actual transaction that contains regular hash. Once that happens hash is provided to such transations. Hope it makes sense why it's ok of there's no `fromAddress` as then it just falls back to regular transaction link by `hash`.

File: src/utils/blockchainExplorer.js [hunk 3]
```diff
@@ -38,7 +52,21 @@ export function viewOnBlockchain(hash: ?string) {
     return;
   }
 
-  const explorerLink = smartWalletService.getConnectedAccountTransactionExplorerLink(hash);
+  let explorerLink;
+  let fromAccount;
+
+  if (fromAddress) {
```
[the remark this answers is not in the document]
Comment (poocart) [reply src/utils/blockchainExplorer.js:58]: Yes. Transactions with non Account related `fromAddress` are any kind of transactions that are imported and have regular hash which is provided to Etherspot SDK explorer which provides generic explorer link. Hope it makes sense.

File: src/screens/History/History.js [hunk 4]
```diff
@@ -38,15 +41,15 @@ function HistoryScreen() {
   const { t } = useTranslationWithPrefix('history');
   const navigation = useNavigation();
 
-  const walletType = useSmartWalletType();
+  const activeAccount = useActiveAccount();
```
Comment (mdjastrzebski) [inline src/screens/History/History.js:44]: 👍🏻

File: src/selectors/archanova.js [hunk 5]
```diff
@@ -104,30 +109,17 @@ export const useSmartWalletStatus = (): ArchanovaWalletStatus => {
...
-  ARCHANOVA: ('archanova': 'archanova'),
-};
-
-export type SmartWalletType = $Values<typeof SMART_WALLET_TYPE>;
-
-export const useSmartWalletType = (): SmartWalletType => {
-  return SMART_WALLET_TYPE.ARCHANOVA;
-};
-
 export const useSupportedChains = (): Chain[] => {
```
Comment (mdjastrzebski) [inline src/selectors/archanova.js:112]: Maybe we should move it to `utils/chains` or something, as this is not archanova related

File: src/selectors/selectors.js [hunk 6]
```diff
@@ -104,3 +108,5 @@ export const sablierSelector = ({ sablier }: RootReducerState) => sablier;
 export const rariSelector = ({ rari }: RootReducerState) => rari;
 
 export const liquidityPoolsSelector = ({ liquidityPools }: RootReducerState) => liquidityPools;
+
+export const useActiveAccount = (): ?Account => useRootSelector(activeAccountSelector);
```
Comment (mdjastrzebski) [inline src/selectors/selectors.js:112]: Awesome

**Review by mdjastrzebski — APPROVED** [review]

**Review by mdjastrzebski — APPROVED** [review]

**Pull request merged**